app/src/interfaces/_system/system-raw-editor/system-raw-editor.vue

```diff
@@ -133,10 +133,20 @@ watch(
 		const currentValue = codemirror?.getValue();
 
 		if (currentValue !== newValue) {
-			codemirror?.setValue(unref(isObjectLike) ? JSON.stringify(newValue, null, 4) : String(newValue ?? ''));
+			codemirror?.setValue(getStringifiedValue(newValue));
 		}
 	}
 );
+
+function getStringifiedValue(value: string | object | undefined) {
+	if (unref(isObjectLike)) {
+		const valueToStringify = typeof value === 'string' && isValidJSON(String(value)) ? parseJSON(String(value)) : value;
+
+		return JSON.stringify(valueToStringify, null, 4);
+	} else {
+		return String(value ?? '');
+	}
+}
```

Comment:
Since we're more or less using the same logic for all three interfaces, I'm wondering if we should place it in a reusable util instead.

```ts
// something like this
function getStringifiedValue(value: string | object | undefined, isObject: boolean) {
	if (isObject) {
		const valueToStringify = typeof value === 'string' && isValidJSON(String(value)) ? parseJSON(String(value)) : value;
		return JSON.stringify(valueToStringify, null, 4);
	} else {
		return String(value ?? '');
	}
}

// usage in system-raw-editor.vue like this
getStringifiedValue(newValue, unref(isObjectLike)

// usage in form-field-raw-editor.vue like this
getStringifiedValue(props.currentValue, type.value === 'object')
```

Would also make the spacing (`4` vs. `\t`) in the `JSON.stringify` calls consistent. Unless I'm missing the reason why this needs to be different 🤔

Reply:
Makes sense! Updated it to a reusable util as you described 👍

Reply:
Awesome work, even with tests ❤️